api/insight/apimiddleware.go

```diff
@@ -69,13 +72,24 @@ func (c *insightApiContext) getBlockHashCtx(r *http.Request) (string, error) {
 
 // GetRawHexTx retrieves the ctxRawHexTx data from the request context. If not
 // set, the return value is an empty string.
-func (c *insightApiContext) GetRawHexTx(r *http.Request) (string, bool) {
+func (c *insightApiContext) GetRawHexTx(r *http.Request) (string, error) {
```

Comment:
Actually please delete this method and modify the `GetRawHexTx` function in the `middleware` package.  The receiver on this method is completely unnecessary.

Reply:
To clarify, this also means moving PostBroadcastTxCtx to `middleware` too because `middleware.ctxRawHexTx` needs to be set.